### ISSUE TITLE
Allow deletion of original source image in Android `ImagePicker`

### ISSUE THREAD

**Issue by nikki93** [expo/expo#449]
The Android `ImagePicker` implementation currently copies the generated image to an app-accessible place in the Expo file system. The original copy is still kept around and there is no option for it to be deleted. Users may desire the deletion of the original copy as expressed, for example, in this forum post: https://forums.expo.io/t/image-picker-deleting-the-new-photo/2003

It may make sense to expose an option that specifies whether the original copy should be deleted. I am yet unsure what the default for that option would be. Feel free to weigh in!

**Comment by maximilian-ruppert** [comment]
Hi, I think this would be a great addition to the `ImagePicker`. In my opinion the default should be, that the original copy should be deleted, because it is easy to oversee that it actually got created in the first place.

**Comment by caedmonm** [comment]
Given that iOS doesn't store an image elsewhere it would make sense to me if the default was for it to delete the original.

**Comment by caedmonm** [comment]
Any new on this issue, I notice @terribleben removed the sdk: 21 label, not sure if that's an ominous sign or not?

**Comment by maximilian-lindsey** [comment]
I still have the problem with SDK 23. When I take a picture with the image picker, one image get's stored in my Expo App and one directly in my phones image directory. To make it worse, the one in expo seems to have a smaller resolution than the on in my default camera directory. (Running Android 7.0)

**Comment by caedmonm** [comment]
Yes, still the same problem here too although I hadn't noticed the resolution problem. It's a bit of a security issue for the app I'm working on so really hope a fix can be found soon...

**Comment by maximilian-lindsey** [comment]
@caedmonm you could try using the camera component expo offers, but one thing I don't like about it, is the fact that I have to build a ui (things like a flash toggle...) for it myself instead of using the native camera ui.

**Comment by maximilian-lindsey** [comment]
@nikki93 I still have this problem using SDK 24. Every picture I take with the ImagePicker, is created in the cache and additionally as a photo in my regular camera role on my Moto G4 (Android 7). The picture in the apps cache has only the dimensions of my screen (864x1152) and the photo in my camera roll has normal dimensions (3456x4608).

**Comment by caedmonm** [comment]
Thanks @maximilian-lindsey - that's exactly what I'm doing but yes, seems unnecessary, I'd be happy to stick with the built-in UI and save myself some time.

**Comment by yehudahkay** [comment]
+1

**Comment by chuanxie** [comment]
@nikki93  any updated on this? on Android.
because it will save so much time to delete the images from gallery, The name it copy from cache to gallery is not the same.  very difficult to remove it programmatically. 
Thanks a lot

**Comment by chuanxie** [comment]
@nikki93 Hi nikki, could @elimirks 's fixes to be included in the next release of expo? cheers

**Comment by aalices** [comment]
I'm going to prevent the images being saved to gallery as it is how it works on iOS. They will be saved in app's cache directory and if one wants to move the pic to the gallery they can use `CameraRoll` or `MediaLibrary` coming in new SDK. not sure if this change will be available with upcoming release, will let you guys know!
